test/test.js

```diff
@@ -122,3 +125,64 @@ describe('Web pages ',function(){
     });
 });
 
+describe('Google Calendar API', function() {
+    var path = 'http://localhost:8080/calendar/event'
+    var eventForm =
+    {   title: 'Judgement Day',
+        description: '',
+        location: 'Cheyenne Mountain, Colorado, USA',
+        details: 'Skynet becomes self aware',
+        startDate: '1997-08-29',
+        startHour: '02',
+        startMinute: '14',
+        endDate: '1997-08-29',
+        endHour: '23',
+        endMinute: '59',
+        ssid: 'Artificial Intelligence'
+    }
+    var modifyEvent =
+    '&title='         + eventForm.title           +
+    '&description='   + eventForm.description     +
+    '&location='      + eventForm.location        +
+    '&details='       + eventForm.details         +
+    '&startDate='     + eventForm.startDate       +
+    '&endDate='       + eventForm.endDate         +
+    '&startHour='     + eventForm.startHour       +
+    '&endHour='       + eventForm.endHour         +
+    '&startMinute='   + eventForm.startMinute     +
+    '&endMinute='     + eventForm.endMinute       +
+    '&ssid='           + eventForm.ssid;
+
+    describe('Event insertion.', function() {
+        it ('Creates and inserts an event into the Calendar.', function(done) {
+            request.post({url: path, form: eventForm}, function(error, response, body) {
+                expect(response.statusCode).to.equal(302);
+                done();
+            });
+        });
+    });
+
+    describe('Event retreival, modification, deletion.', function() {
+        var startDate = eventForm.startDate + 'T' + eventForm.startHour + ':' + eventForm.startMinute + ':00-00:00';
+        var endDate = eventForm.endDate + 'T' + eventForm.endHour + ':' + eventForm.endMinute + ':00-00:00';
+        var getEvent = '?startDate=' + startDate + '&endDate=' + endDate;
+        it ('Retreives, modifies, and deletes the inserted event.', function(done) {
```

Comment:
You can use the functions before and after to work around those problems. For example in the test where you add an even you can use the after function to delete it. And then for testing the modification and deletion you can have a beforeEach that adds a seperate event and stores the id  and an afterEach function that cleans up after the modification.